File: 0200-number-of-islands/0200-number-of-islands.js
/**
 * @param {character[][]} grid
 * @return {number}
 */


var numIslands = function(grid) {
    let count=0;
    
function calculate(i,j){
    if (i < 0 || j < 0 || i >= grid.length || j >= grid[0].length || grid[i][j] === '0') {
        return; 
    }
  
  grid[i][j]="0";
  calculate(i+1,j);
  calculate(i,j+1);
  calculate(i-1,j);
  calculate(i,j-1);


}

for(let i=0;i<grid.length;i++){
    for(let j=0;j<grid[0].length;j++){
        if(grid[i][j]==="1"){
            calculate(i,j);
            count++;


        }
    }
}


return count;
    
};

File: 0200-number-of-islands/0200-number-of-islands.cpp
class Solution {
public:

    void dfs(vector<vector<char>>& grid,int i,int j){
        int n=grid.size();
        int m=grid[0].size();
        if(i<0 || i>=n)return;
        if(j<0 || j>=m)return;


        if(grid[i][j]=='0')return ;

        if(grid[i][j]=='1')grid[i][j]='0';
        dfs(grid,i+1,j); 
         dfs(grid,i-1,j); 
         dfs(grid,i,j+1);
         dfs(grid,i,j-1);
    

    }
    int numIslands(vector<vector<char>>& grid) {

        int n=grid.size();
        int m=grid[0].size();

            int result=0;
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(grid[i][j]=='1'){
                    result++;
                    dfs(grid,i,j);
                }
            }
        }

            return result;    
    }
};